core/schema/env.graphqls

```diff
@@ -0,0 +1,98 @@
+extend type Query {
+  "Load a environment from ID."
+  environment(id: EnvironmentID): Environment!
+
+  "Load a environment check from ID."
+  environmentCheck(id: CheckID): Check!
+
+  "Return the current environment being executed in."
+  currentEnvironment: Environment!
+}
+
+"A unique environment identifier."
+scalar EnvironmentID
+
+"A unique environment check identifier."
+scalar CheckID
+
+"A unique environment check result identifier."
+scalar CheckResultID
+
+"A group of Dagger entrypoints that can be queried and/or invoked."
+type Environment {
+  "A unique identifier for this environment."
+  id: EnvironmentID!
+
+  "Initialize this environment from the given directory and config path"
+  load(source: DirectoryID!, configPath: String!): Environment!
+
+  "Name of the environment"
+  name: String!
+
+  "This environment plus the given check"
+  withCheck(id: CheckID!): Environment!
+
+  "The list of checks in this environment"
+  checks: [Check!]
+
+  "The check in this environment with the given name, if any"
+  check(name: String!): Check
+}
+
+"An entrypoint for tests, lints or anything that can pass/fail."
+type Check {
```

Comment:
I also left out the previous APIs for check flags (i.e. parameters to checks) for now given our most recent conversation on the topic. This is another case where I think it's worth starting here and seeing how far we can get without them before considering re-adding.